File: tests/integration/call_component_failures/nodejs/component.ts
// Copyright 2016-2024, KhulnaSoft Ltd.  All rights reserved.

import * as codeinfra from "@codeinfra/codeinfra";

interface ComponentArgs {
    foo: codeinfra.Input<string>;
}

export class Component extends codeinfra.ComponentResource {
    public readonly foo!: codeinfra.Output<string>;

    constructor(name: string, args: ComponentArgs, opts?: codeinfra.ComponentResourceOptions) {
        const inputs: any = {};
        inputs["foo"] = args.foo;
        super("testcomponent:index:Component", name, inputs, opts, true);
    }

    getMessage(args: Component.GetMessageArgs): codeinfra.Output<Component.GetMessageResult> {
        return codeinfra.runtime.call("testcomponent:index:Component/getMessage", {
            "__self__": this,
            "echo": args.echo,
        }, this);
    }
}

export namespace Component {
    export interface GetMessageArgs {
        echo: codeinfra.Input<string>;
    }

    export interface GetMessageResult {
        message: string;
    }
}

File: tests/integration/call_component_failures/python/component.py
# Copyright 2016-2024, KhulnaSoft Ltd.  All rights reserved.

from typing import Optional

import codeinfra

class Component(codeinfra.ComponentResource):
    foo: codeinfra.Output[str]

    def __init__(self, name: str, foo: codeinfra.Input[str], opts: Optional[codeinfra.ResourceOptions] = None):
        props = dict()
        props["foo"] = foo
        super().__init__("testcomponent:index:Component", name, props, opts, True)

    def get_message(self) -> codeinfra.Output[str]:
        __args__ = dict()
        __args__['__self__'] = self
        return codeinfra.runtime.call('testcomponent:index:Component/getMessage', __args__, res=self)
